NuRadioReco/modules/io/coreas/coreas.py

```diff
@@ -123,136 +636,382 @@ def calculate_simulation_weights(positions, zenith, azimuth, site='summit', debu
+            obs_positions_onsky.append(efield.get_position_onsky())
+            electric_field_on_sky.append(efield.get_trace().T)
+            efield_times.append(efield.get_times())
+
+        # shape: (n_observers, n_samples, (eR, eTheta, ePhi))
+        self.electric_field_on_sky = np.array(electric_field_on_sky)
+        self.efield_times = np.array(efield_times)
+        self.sampling_rate = 1. / (self.efield_times[0][1] - self.efield_times[0][0])
+        self.obs_positions_geo = np.array(obs_positions_geo)
+        self.obs_positions_vxB_vxvxB = np.array(obs_positions_onsky)
+
+        self.max_coreas_efield = np.max(np.abs(self.electric_field_on_sky))
+        self.empty_efield = np.zeros_like(self.electric_field_on_sky[0, :, :])
+
+        self.star_radius = np.max(np.linalg.norm(self.obs_positions_vxB_vxvxB[:, :-1], axis=-1))
+        self.geo_star_radius = np.max(np.linalg.norm(self.obs_positions_geo[:-1, :], axis=0))
+        logger.info(
+            f'Initialize star shape pattern for interpolation. '
+            f'The shower arrives at zenith={self.zenith / units.deg:.0f}deg, '
+            f'azimuth={self.azimuth / units.deg:.0f}deg with radius {self.star_radius:.0f}m in the shower plane '
+            f'and {self.geo_star_radius:.0f}m on ground. ')
+        self.star_shape_initialized = True
+
+    def get_sampling_rate(self):
+        """
+        returns the sampling rate of the electric field
+        """
+        return self.sampling_rate
+
+    def get_empty_efield(self):
+        """
+        returns the an array of zeros in the shape of the electric field on the sky
+        """
+        if not self.star_shape_initialized:
+            logger.error('interpolator not initialized, call initialize_star_shape first')
+            return None
+        else:
+            return self.empty_efield
+
+    def get_max_efield(self):
+        """
+        returns the maximum value of the electric field provided by coreas
+        """
+        if not self.star_shape_initialized:
+            logger.error('interpolator not initialized, call initialize_star_shape first')
+            return None
+        else:
+            return self.max_coreas_efield
+
+    def get_star_radius(self):
+        """
+        returns the maximal radius of the star shape pattern in the shower plane
+        """
+        if not self.star_shape_initialized:
+            logger.error('interpolator not initialized, call initialize_star_shape first')
+            return None
+        else:
+            return self.star_radius
+
+    def get_geo_star_radius(self):
+        """
+        returns the maximal radius of the star shape pattern on ground
+        """
+        if not self.star_shape_initialized:
+            logger.error('interpolator not initialized, call initialize_star_shape first')
+            return None
+        else:
+            return self.geo_star_radius
+
+    def initialize_efield_interpolator(self, interp_lowfreq, interp_highfreq):
```

Comment:
With the LOFAR version we had an optional parameter for the interpolation parameters. I think that was a clean solution to have some flexibility for the user. What do you think?

Reply:
Can you post a link to the LOFAR interpolation?